department-of-veterans-affairs/caseflow
# Justin/appeals 46859

Resolves [Correspondence Model Has correspondence_type Set Up to Populate from CMP](https://jira.devops.va.gov/browse/APPEALS-46859)

# Description
Removed requirement for correspondence_type_id to be assigned, and updated spec test files to not assign correspondence_type_id

## Acceptance Criteria
- [ ] The correspondence model should not require correspondence_type_id to be assigned upon creation. 
- [ ] This needs to be reflected in the correspondence seed data as well.

Correspondence is now able to be added when the correspondence_type_id is nil. 
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/95875751/9dd2ca24-e83d-4058-9570-1ef58f7f8c66)



## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

1. Start up a backend server in a local development environment. This can be done in terminal within the appeals/caseflow directory using the command `make run-backend` or `make run-backend-m1` on newer Macs.
2. Start a rails server. This can be done within a new terminal tab/window within the appeals/caseflow directory using the command `make c`
3. Paste the command `veteran = Veteran.first` into the rails terminal once it has successfully started. Press enter to run this command.
4. Next, paste the command `Correspondence.create!(
          uuid: SecureRandom.uuid,
          portal_entry_date: Time.zone.now,
          source_type: "Mail",
          package_document_type_id: 15,
          correspondence_type_id: nil,
          cmp_queue_id: 1,
          cmp_packet_number: 12,
          va_date_of_receipt: Faker::Date.between(from: 90.days.ago, to: Time.zone.yesterday),
          notes: "This is a test note",
          veteran_id: veteran.id
        )`, and paste in as one line. Then press enter to run the command.
5. Before this code change, the command would not allow the correspondence to be created, because of the "nil" value in the correspondence_type_id. The expected result now should be that the correspondence is created.
